theseus/core/loss.py

```diff
@@ -23,20 +25,20 @@ def linearize(x: torch.Tensor, raidus: torch.Tensor) -> torch.Tensor:
 class WelschLoss(Loss):
     @staticmethod
     def evaluate(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
-        return radius - radius * torch.exp(-x / radius)
+        return radius - radius * torch.exp(-x / (radius + _LOSS_EPS))
 
     @staticmethod
     def linearize(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
-        return torch.exp(-x / radius)
+        return torch.exp(-x / (radius + _LOSS_EPS))
 
 
 class HuberLoss(Loss):
     @staticmethod
     def evaluate(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
         return torch.where(
-            x > radius, 2 * torch.sqrt(radius * x.max(radius)) - radius, x
+            x > radius, 2 * torch.sqrt(radius * x.max(radius) + _LOSS_EPS) - radius, x
```

Comment:
I think it's fine to just rely on torch for checking if the shapes are good or not. There is no requirement about dimensions, since the arguments are just torch tensors.